library/general/src/lib/ui/srv_status_component.rb

```diff
@@ -0,0 +1,236 @@
+      Yast::UI.ChangeWidget(Id("#{id_prefix}_enabled"), :Value, @enabled)
+      Yast::UI.ReplaceWidget(Id("#{id_prefix}_status"), status_widget)
+    end
+
+    # Reloads the service only if the user requested so. It should be called
+    # after saving the settings.
+    def reload
+      reload_service if service_running? && @reload
+    end
+
+    # rubocop:disable Style/TrivialAccessors
+
+    # Checks if the user requested the service to be enabled on boot
+    #
+    # @return [Boolean]
+    def enabled?
+      @enabled
+    end
+
+    # rubocop:enable Style/TrivialAccessors
+
+    # Content for the help
+    def help
+      _(
+        "<p><b><big>Current status</big></b><br>\n"\
+        "Displays the current status of the service. The status will remain "\
+        "the same after saving the settings, independently of the value of "\
+        "'start service during boot'.</p>\n"\
+        "<p><b><big>Reload After Saving Settings</big></b><br>\n"\
+        "Only applicable if the service is currently running. "\
+        "Ensures the running service reloads the new configuration after "\
+        "saving it (via 'ok' or 'save' buttons).</p>\n"\
+        "<p><b><big>Start During System Boot</big></b><br>\n"\
+        "Check this field to enable the service at system boot. "\
+        "Un-check it to disable the service. "\
+        "This does not affect the current status of the service in the already "\
+        "running system.</p>\n"
+      )
+    end
+
+  protected
+
+    attr_reader :id_prefix
+
+    # Checks if the service is currently running
+    #
+    # Must be redefined for services not following standard procedures
+    #
+    # @return [Boolean]
+    def service_running?
+      Yast::Service.active?(@service_name)
+    end
+
+    # Checks if the service is currently enabled on boot
+    #
+    # Must be redefined for services not following standard procedures
+    #
+    # @return [Boolean]
+    def service_enabled?
+      Yast::Service.enabled?(@service_name)
+    end
+
+    # Starts the service inmediatly
+    #
+    # Must be redefined for services not following standard procedures
+    def start_service
+      log.info "Default implementation of SrvStatusComponent#start_service for #{@service_name}"
+      Yast::Service.Start(@service_name)
+    end
+
+    # Stops the service inmediatly
+    #
+    # Must be redefined for services not following standard procedures
+    def stop_service
+      log.info "Default implementation of SrvStatusComponent#stop_service for #{@service_name}"
+      Yast::Service.Stop(@service_name)
+    end
+
+    # Reloads the configuration of a running service
+    #
+    # Must be redefined for services not following standard procedures
+    def reload_service
+      log.info "Default implementation of SrvStatusComponent#reload_service for #{@service_name}"
+      Yast::Service.Reload(@service_name)
+    end
```

Comment:
again here I expect usage of `@service` object, so you do not need to use redefinition for non-standard services :(